Source/WebCore/html/parser/HTMLConstructionSite.h

```diff
@@ -73,6 +74,10 @@ struct HTMLConstructionSiteTask {
         return downcast<ContainerNode>(child.get());
     }
 
+    Ref<ContainerNode> protectedNonNullParent() const { return *parent; }
+    Ref<Node> protectedNonNullChild() const { return *child; }
+    Ref<Node> protectedNonNullNextChild() const { return *nextChild; }
```

Comment:
Should these ASSERT that their members are non-null?

Reply:
Good idea. will update.

Reply:
Actually, this is not needed because RefPtr's operator*() already ASSERTs this.

```
T& operator*() const { ASSERT(m_ptr); return *PtrTraits::unwrap(m_ptr); }
```